src/bucket.ts

```diff
@@ -323,6 +324,16 @@ export interface UploadOptions
   resumable?: boolean;
 }
 
+export interface UploadDirectoryOptions extends CreateWriteStreamOptions {
+  recurse?: boolean;
+}
+
+export interface UploadDirectoryCallback {
+  (err: Error | null, resp?: Array<{[k: string]: string | Error}>): void;
```

Comment:
Can we move the nested object to its own type?

Reply:
Done. Left the types as is till we finalize what this method should return.